tests/blivetutils_tests/blivetutilstestcase.py

```diff
@@ -0,0 +1,70 @@
+import glob
+import os
+import re
+import subprocess
+import time
+import unittest
+
+from blivetgui.blivet_utils import BlivetUtils
+
+
+def run_command(command):
+    res = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
+                           stderr=subprocess.PIPE)
+
+    out, err = res.communicate()
+    if res.returncode != 0:
+        output = out.decode().strip() + "\n\n" + err.decode().strip()
+    else:
+        output = out.decode().strip()
+    return (res.returncode, output)
+
+
+class BlivetUtilsTestToolkit(unittest.TestCase):
+
+    vdevs = None
+    blivet_utils = None
+
+    def setUp(self):
+        self.blivet_utils = BlivetUtils(exclusive_disks=self.vdevs)
+
+        # check we are really working only with the targetcli disks
+        disks = self.blivet_utils.get_disks()
+        self.assertCountEqual(self.vdevs, [d.name for d in disks])
+
+    def get_blivet_device(self, device_name):
+        return self.blivet_utils.storage.devicetree.get_device_by_name(device_name)
+
+
+@unittest.skipUnless(os.geteuid() == 0, "requires root access")
+class BlivetUtilsTestCase(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+
+        orig_devs = {dev for dev in os.listdir("/dev") if re.match(r"sd[a-z]+$", dev)}
+
+        test_dir = os.path.abspath(os.path.dirname(__file__))
+        ret, _out = run_command("targetcli restoreconfig %s" % os.path.join(test_dir, "targetcli_config.json"))
+        assert ret == 0
+        time.sleep(0.1)
```

Comment:
This should probably use ``udevadm settle`` instead.

Reply:
Fixed